src/firefly/java/edu/caltech/ipac/firefly/server/ws/Workspaces.java

```diff
@@ -0,0 +1,81 @@
+package edu.caltech.ipac.firefly.server.ws;
+
+import edu.caltech.ipac.firefly.data.WspaceMeta;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Define workspaces API that will be used by UI client to connect to WS
+ * To ease the interaction between client and server, the interface is defined with relative path/uri to remote ws home
+ * home dir / root url <b>should/expected to</b> be defined in implementation (see {@link edu.caltech.ipac.firefly.server.WorkspaceManager.PROPS}
+ * <p>
+ * Created by ejoliet on 6/16/17.
+ */
+public interface Workspaces {
+
+    /**
+     * Gets files from relative path
+     *
+     * @param parentWspaceRelDir ws parent directory to get list of files
+     * @param depth              if -1, give all the folders/files below the parent, 0 return the parent, 1 return children only
+     * @return {@link WsResponse} - list of {@link WspaceMeta} can be found in response, see {@link WsResponse#getWspaceMeta()} resource list in response object}
+     * @throws IOException handle ws error
+     */
+    WsResponse getList(String parentWspaceRelDir, int depth) throws IOException;
+
+    /**
+     * Get file relative to ws home
+     *
+     * @param wspaceRelFile remote URI relative file path , i.e. "x/y/z/aaa.tbl" will be located under WS_HOME at WS_URL
+     * @param outputFile    local file downloaded
+     * @return {@link WsResponse} response string should contain the abs path of the file TODO could contain more metadata
+     * @throws WsException handle ws error
+     */
+    WsResponse getFile(String wspaceRelFile, File outputFile) throws WsException;
```

Comment:
yes, i think is used in producer/consumer pattern but we can add wsexceptions and more workspace related. I will add the method for now and no implementation, how about that?